spec/controllers/tickets_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TicketsController, type: :controller do
+  let!(:user) { FactoryBot.create :user }
+  let!(:event) { FactoryBot.create :event }
+  before do
+    allow(request.env['warden']).to receive(:authenticate!).and_return(user)
```

Comment:
Ja korzystam zawsze z czegoś takiego
https://github.com/plataformatec/devise/wiki/How-To:-sign-in-and-out-a-user-in-Request-type-specs-(specs-tagged-with-type:-:request)

Wydaje mi się to wygodniejsze w użyciu i bardziej odwzorowuje prawdziwe logowanie. Tzn. nie tylko metoda `current_user` jest dostępna, ale też `user_signed_in?`itp. Nie trzeba mockować ręcznie